bakerydemo/templates/blog/blog_index_page.html

```diff
@@ -1,22 +1,55 @@
 {% extends "base.html" %}
-{% load wagtailimages_tags %}
+{% load wagtailcore_tags navigation_tags wagtailimages_tags %}
 
 {% block content %}
-    {% include "base/include/header.html" %}
+    {% include "base/include/header-index.html" %}
 
     <div class="container">
+    {% if tag %}
         <div class="row">
-            <div class="col-md-7 col-md-offset-2">
-                {% if tag %}
-                    <h3>Posts tagged with "{{ tag }}":</h3>
-                {% endif %}
-                {% for blog in blogs %}
-                    <div>
-                        <h2><a href="{{ blog.url }}">{{ blog.title }}</a></h2>
-                        {{ blog.body|truncatewords_html:10 }}
+            <div class="col-md-12">
+                <p>Viewing all blog posts by <span class="outline">{{ tag }}</span></p>
+            </div>
+        </div>
+    {% endif %}
+
+    {% if page.get_child_tags %}
+        <ul class="blog-tags tags list-inline">
+            {% for tag in page.get_child_tags %}
+                <li><a href="{{ tag.url }}">{{ tag }}</a></li>
+            {% endfor %}
+        </ul>
+    {% endif %}
+
+        <div class="row row-eq-height blog-list">
+        {% if blogs %}
+            {% for blog in blogs %}
+                <li class="col-xs-12 col-sm-6 col-md-3 blog-list-item">
+                <a href="{% pageurl blog %}">
+                    <div class="image">
+                        {% image blog.image fill-850x450-c50 as image %}
+                        <img src="{{ image.url }}" width="{{ image.width }}" height="{{ image.height }}" alt="{{ image.alt }}" class="" />
+                    </div>
+                    <div class="text">
+                        <h2 class="blog-list-title">{{ blog.title }}</h2>
+                        <p>{{ blog.introduction|truncatewords:15 }}</p>
```

Comment:
Normally on a production site where we show introductions both on the listview and detailview we have two separate fields for this reason, since the introduction on the detailview can be substantially longer than on the listview. Personally I think here it's okay to just have the single field with the check on the blog index page, though totally open to other solutions.